xbmc/music/MusicDatabase.cpp

```diff
@@ -3823,20 +3817,36 @@ bool CMusicDatabase::GetSongsFullByWhere(const std::string &baseDir, const Filte
 
     // Apply the limiting directly here if there's no special sorting but limiting
     bool limited = extFilter.limit.empty() && sortDescription.sortBy == SortByNone &&
-                   (sortDescription.limitStart > 0 || sortDescription.limitEnd > 0);
+      (sortDescription.limitStart > 0 || sortDescription.limitEnd > 0);
     if (limited)
       strSQLExtra += DatabaseUtils::BuildLimitClause(sortDescription.limitEnd, sortDescription.limitStart);
 
+    std::string strSQL;
     if (!artistData)
-      strSQL = PrepareSQL(strSQL, !filter.fields.empty() && filter.fields.compare("*") != 0 ? filter.fields.c_str() : "songview.* ") + strSQLExtra;
+      strSQL = "SELECT songview.* FROM songview " + strSQLExtra;
     else
-    {
-      strSQL = PrepareSQL(strSQL, !filter.fields.empty() && filter.fields.compare("*") != 0 ? filter.fields.c_str() : "songview.*, songartistview.* ");
+    { // Get data from song and song_artist tables to fully populate songs with artists
```

Comment:
Finicky is fine. I did that layout to put the non join SQL at the top, and broke the else-if so that I could comment on both join SQL in one place. Made sense at the time.